SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -69,6 +72,16 @@ public void runActionOnPoll() {
         continue;
       }
 
+      if (start - taskId.getStartedAt() >= configuration.getTaskExecutionTimeout()) {
```

Comment:
Follow up to this. Should this conditional have the work of ignoring long running request types or should we have the validator make sure that the config field `taskExecutionTimeLimitMillis` is only set for non-long running tasks? [likely here](https://github.com/HubSpot/Singularity/blob/master/SingularityService/src/main/java/com/hubspot/singularity/data/SingularityValidator.java#L202-L205)? [maybe even both](http://porquenolosdos.com/)?